[Excel] Release ExcelApi 1.18 to GA

alison-mk opened the pull request:
Will run GenerateDocs after https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71857 is approved and merged

ElizabethSamuel-MSFT reviewed (commented):
Overall, looks good. But I'm not seeing any 1.18 entry on the Excel requirement sets overview page (https://review.learn.microsoft.com/en-us/javascript/api/requirement-sets/excel/excel-api-requirement-sets?view=common-js-preview&branch=pr-en-us-2198)

ElizabethSamuel-MSFT reviewed (approved):
Approving but if omission noted in my comment wasn't intentional, please do fix before merging

alison-mk commented:
@ElizabethSamuel-MSFT Thanks for catching that! I missed it--I've added it to our internal process doc to help avoid that in the future. I've updated the build table.
